src/lib/proxy-server.ts

```diff
@@ -89,25 +87,15 @@ export async function startProxyServer(): Promise< boolean > {
 					resolve();
 				} )
 				.on( 'error', ( err ) => {
-					console.log( err );
-					console.error( `Error starting proxy server on port 80:`, err );
+					console.error( `Error starting proxy server on port 80` );
 					reject( err );
 				} );
 		} );
 
 		return true;
 	} catch ( error ) {
 		if ( error instanceof Error && 'code' in error && error.code === 'EADDRINUSE' ) {
-			const mainWindow = await getMainWindow();
-			dialog.showMessageBox( mainWindow, {
-				type: 'error',
-				message: __( 'Custom domain set up failed' ),
-				detail: __(
-					'Studio needs to use port 80, but it’s already in use by another app. Close any local development apps and restart Studio.'
-				),
-				buttons: [ __( 'OK' ) ],
-			} );
-			return false;
+			throw new Error( 'Studio failed to start the proxy server' );
```

Comment:
Let's move the part responsible for dialog to the renderer.

Reply:
It looks a bit odd that we throw a new exception for known error types but just `return false` for unknown error types. The known error types indicate that port 80 is busy, but other errors also indicate that Studio failed to start the proxy server.

Reply:
I've updated that logic to handle error cases only through exceptions and cleaned up the logic around `false`.

Reply:
Nice improvement 👍